ADALiOS/ADALiOS/ADTokenCacheItem.m

```diff
@@ -169,4 +186,48 @@ - (NSString*)description
             [NSString adIsStringNilOrBlank:_refreshToken] ? @"(nil)" : @"(present)", _resource];
 }
 
+- (NSString *)clientId
+{
+    return _clientId;
+}
+
+- (void)setClientId:(NSString *)clientId
+{
+    _clientId = clientId;
+    [self calculateHash];
+}
+
+- (ADUserInformation *)userInformation
+{
+    return _userInformation;
+}
+
+- (void)setUserInformation:(ADUserInformation *)userInformation
+{
+    _userInformation = userInformation;
+    [self calculateHash];
```

Comment:
calculateHash is expensive and -hash gets called a lot when using Foundation Collections. -hash is supposed to be a fast call.